rest/route/github.go

```diff
@@ -130,17 +130,16 @@ func (gh *githubHookApi) shouldSkipWebhook(ctx context.Context, owner, repo stri
 	return (fromApp && !hasApp) || (!fromApp && hasApp)
 }
 
-func (gh *githubHookApi) Run(_ context.Context) gimlet.Responder {
+func (gh *githubHookApi) Run(ctx context.Context) gimlet.Responder {
 	// GitHub occasionally aborts requests early before we are able to complete the full operation
-	// (for example enqueueing a PR to the commit queue). We therefore want to use a custom context
-	// instead of using the request context.
-	ctx, cancel := context.WithTimeout(context.Background(), githubWebhookTimeout)
+	// (for example enqueueing a PR to the commit queue). We therefore want to use a custom timeout.
+	newCtx, cancel := context.WithTimeout(ctx, githubWebhookTimeout)
```

Comment:
We should use context.WithoutCancel on the ctx too
```suggestion
	newCtx, cancel := context.WithTimeout(context.WithoutCancel(ctx), githubWebhookTimeout)
```
The comment should reflect it a bit, maybe like `We therefore ignore if the given context is canceled and apply our own to prevent hanging requests`